src/kinematics_plugin.cpp

```diff
@@ -253,6 +253,10 @@ struct BioIKKinematicsPlugin : kinematics::KinematicsBase {
                 std::string("bio2_memetic"));
     lookupParam("counter", ikparams.enable_counter, false);
     lookupParam("threads", ikparams.thread_count, 0);
+    ros::NodeHandle nodeHandle("~");
+    int random_seed;
+    if (nodeHandle.getParam("random_seed", random_seed))
+      ikparams.random_seed = random_seed;
```

Comment:
This is buggy. Please use `lookupParam`. all kinematics plugins should use this method over creating nodehandles.
```suggestion
    lookupParam("random_seed", ikparams.random_seed, std::random_device()());
```

Reply:
That does not compile as `ikparams.random_seed` is of type `std::minstd_rand::result_type` and `nodeHandle.param` has no overload for that. We could use:
```
random_seed = nodeHandle.param<int>("random_seed", random_seed);
```

Bot honestly I'm wondering which ROS version you target, `getParam()` is there since the beginning of ros_comm.git (2010), or am I missing something else? (At least it works fine with Melodic here).

Reply:
yes, I wrote the suggestion without testing.
It's not about the ROS version, but about [the additional fallback behavior](https://github.com/ros-planning/moveit/blob/master/moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h#L621-L649) to handle the common case that plugin parameters are defined in the `kinematics.yaml`.

Reply:
Interesting concept.. anyway, should be fixed now.